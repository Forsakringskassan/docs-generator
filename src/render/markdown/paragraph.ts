import type MarkdownIt from "markdown-it";

export function paragraph(): (md: MarkdownIt) => void {
    const renderToken: MarkdownIt.Renderer.RenderRule = (...args) => {
        const [tokens, idx, options, , self] = args;
        return self.renderToken(tokens, idx, options);
    };

    return function (md: MarkdownIt): void {
        /* eslint-disable camelcase -- property is defined in upstream library */
        const renderRule = md.renderer.rules.paragraph_open ?? renderToken;
        md.renderer.rules.paragraph_open = (tokens, idx, index, env, self) => {
            tokens[idx].attrJoin("class", "docs-paragraph");
            return renderRule(tokens, idx, index, env, self);
        };
        /* eslint-enable camelcase */
    };
}
