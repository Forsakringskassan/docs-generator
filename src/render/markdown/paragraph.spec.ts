import markdownIt from "markdown-it";
import { paragraph } from "./paragraph";

const md = markdownIt();
md.use(paragraph());

it("should render paragraph with class", () => {
    expect.assertions(1);
    const markup = md.render(`
lorem ipsum
dolor sit amet

foo bar baz
spam ham
`);
    expect(markup).toMatchSnapshot();
});

it("should not add paragraph to list items", () => {
    expect.assertions(1);
    const markup = md.render(`
- foo
- bar
- baz

lorem ipsum

* spam
* ham
`);
    expect(markup).toMatchSnapshot();
});
