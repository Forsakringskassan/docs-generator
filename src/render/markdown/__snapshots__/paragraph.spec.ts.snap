// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not add paragraph to list items 1`] = `
"<ul>
<li>foo</li>
<li>bar</li>
<li>baz</li>
</ul>
<p class="docs-paragraph">lorem ipsum</p>
<ul>
<li>spam</li>
<li>ham</li>
</ul>
"
`;

exports[`should render paragraph with class 1`] = `
"<p class="docs-paragraph">lorem ipsum
dolor sit amet</p>
<p class="docs-paragraph">foo bar baz
spam ham</p>
"
`;
